0028-web-monetization/0028-web-monetization.md

```diff
@@ -30,271 +29,142 @@ Web Monetization is a proposed browser API that uses ILP micropayments to moneti
+  - The browser will generate a fresh UUID (version 4) and use this as the Correlation ID from this point forward. **This Correlation ID MUST be unique per page load**, not per browser, session nor site.
+- The browser passes these payment details to the user's provider (see [Payment Handler Flow](#payment-handler-flow)).
+- The provider resolves the payment pointer and begins to pay. The provider MAY be acting from a different machine from the user. Cookies and session information MUST not be carried with any requests made to resolve the Payment Pointer, with the exception of the Correlation ID.
+  - On the SPSP query to resolve the Payment Pointer, a `Web-Monetization-Id` header is sent, containing the Correlation ID. The server running the web-monetized site may use this to associate future requests by the user with their payments.
+  - With the `destination_account` and `shared_secret` fetched from the SPSP query, a STREAM connection is established. A single STREAM is opened on this connection, and a positive SendMax is set.
+  - The provider SHOULD set their SendMax high enough that it is never reached, and make payment adjustments by limiting throughput.
+- Once the STREAM connection has fulfilled an ILP packet with a non-zero amount, the provider notified the browser, and the browser dispatches a `CustomEvent` on `document`. Payment SHOULD continue.
+  - The user's agent sets `document.monetizationState` to `started`. This MUST occur before the `monetizationstart` event is fired.
+  - The `CustomEvent`'s type is `monetizationstart`. The `CustomEvent`'s `detail` is an object containing the Payment Pointer and the Correlation ID ([specified below](#monetizationstart)).
+  - The user's agent also emits a `monetizationprogress` ([specified below](#monetizationprogress)), corresponding to this first packet. If there are no listeners the event MAY NOT be emitted.
+- Payment continues until the user closes/leaves the page.
+  - The provider MAY decide to stop/start payment at any time, e.g. if the user is idle, backgrounds the page, or instructs the browser to do so.
+  - If the STREAM connection is severed, the provider will redo the SPSP query to the same Payment Pointer as before with the same Correlation ID. The browser MUST NOT re-process the `<meta>` tags.
+  - Each time a packet with a nonzero amount is fulfilled, the provider notifies the browser, and the browser emits a `CustomEvent` on `document`. The `CustomEvent`'s type is `monetizationprogress`. The `CustomEvent`'s `detail` is an object containing the details of the packet ([specified below](#monetizationprogress)). If there are no listeners the event MAY NOT be emitted.
+
+### Payment Handler Flow
+
+When the browser uses an extension to implement Web Monetization, Payment Handlers MAY NOT be used. Otherwise communication to the user's Web Monetization provider SHOULD be implemented with this flow.
```

Comment:
> When the browser uses an extension to implement Web Monetization, Payment Handlers MAY NOT be used

What's the motivation here? Maybe this could be:

"A provider can be implemented as a Payment Handler supporting the 'webmonetization' payment method. (The payment method specification for this payment method is still under development.)"